modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -795,6 +797,10 @@ private IgniteInternalFuture<SnapshotOperationResponse> initLocalSnapshotStartSt
         }
 
         List<Integer> grpIds = new ArrayList<>(F.viewReadOnly(req.groups(), CU::cacheId));
+        List<Integer> comprGrpIds = grpIds.stream().filter(i -> {
```

Comment:
I suggest to use lightweight filter: `F.view(grpIds, id -> {..}`

Reply:
Agree, there is no need to create new List